brainlit/map_neurons/map_neurons.py

```diff
@@ -0,0 +1,380 @@
+            transformed_deriv[i, :] = np.matmul(J, d).T
+
+        return transformed_deriv
+
+
+def compute_derivs(
+    us: np.array,
+    tck: tuple = None,
+    positions: np.array = None,
+    deriv_method: str = "difference",
+) -> np.array:
+    """Estimate derivatives of a sequence of points. Derivatives can be estimated in two ways:
+    - For curves parameterized by scipy's spline API, spline estimation uses scipy's derivative computation
+    - For a sequence of points, the finite-difference method from Svndquist & Veronis 1970 for nonuniform intervals is used.
+
+    Args:
+        us (np.array): Parameter values (in form returned by scipy.interpolate.splprep).
+        tck (tuple): Knots, bspline coefficients, and degree of spline (in form returned by scipy.interpolate.splprep).
+        positions (np.array): nx3 array of positions (for use by difference method).
+        deriv_method (str, optional): Method to use, spline for scipy.interpolate.splev or difference for  . Defaults to "difference".
+
+    Raises:
+        ValueError: If the wrong combination of arguments/deriv_method is used.
+        ValueError: If derivative method is unrecognized.
+
+    Returns:
+        np.array: Derivative estimates at specified positions.
+    """
+    if deriv_method == "spline":
+        if tck is None or positions is not None:
+            raise ValueError(
+                f"When using spline method, positions should be None and tck should not"
+            )
+        derivs = np.array(splev(us, tck, der=1)).T
+    elif deriv_method == "difference":
+        if tck is not None or positions is None:
+            raise ValueError(
+                f"When using difference method, tck should be None and positions should not"
+            )
+        # Sundqvist & Veronis 1970
+        f_im1 = positions[:-2, :]
+        f_i = positions[1:-1, :]
+        f_ip1 = positions[2:, :]
+        hs = np.diff(us)
+        h_im1 = np.expand_dims(hs[:-1], axis=1)
+        h_i = np.expand_dims(hs[1:], axis=1)
+
+        if len(us) >= 3:
+            diffs = (
+                f_ip1
+                - np.multiply((1 - np.divide(h_i, h_im1) ** 2), f_i)
+                - np.multiply(np.divide(h_i, h_im1) ** 2, f_im1)
+            )
+            diffs = np.concatenate(
+                (
+                    [positions[1, :] - positions[0, :]],
+                    diffs,
+                    [positions[-1, :] - positions[-2, :]],
+                ),
+                axis=0,
+            )
+        elif len(us) == 2:
+            diffs = np.array(
+                [positions[1, :] - positions[0, :], positions[-1, :] - positions[-2, :]]
+            )
+        norms = np.linalg.norm(diffs, axis=1)
+        derivs = np.divide(diffs, np.array([norms]).T)
+    else:
+        raise ValueError(f"Invalid deriv_method argument: {deriv_method}")
+
+    return derivs
+
+
+def transform_GeometricGraph(
```

Comment:
lowercase name of function

```suggestion
def transform_geometricgraph(
```

Reply:
changed